polygonizer_dockwidget.py

```diff
@@ -240,6 +241,8 @@ def eventPushButtonRunPolygonizerOnClick(self):
                     # FIXME we need some snapping flexibility, because this can leave slivers
                     if road.geometry().length() < LEG_LENGTH * 2:
                         LEG_LENGTH = road.geometry().length() / 2
+                    elif road.geometry().length() - LEG_LENGTH * 2 < MAX_SNAP_LENGTH:
+                        LEG_LENGTH = road.geometry().length() / 2
```

Comment:
After getting the UI stuff sorted to where a max snap interval was done, the actual logic to do this was even more straight forward than I could have hoped.